pkg/engine/filesystem.go

```diff
@@ -37,7 +45,6 @@ func (e *Engine) ScanFileSystem(ctx context.Context, c sources.FilesystemConfig)
 	if err != nil {
 		return errors.WrapPrefix(err, "could not init filesystem source", 0)
 	}
-	fileSystemSource.WithFilter(c.Filter)
```

Comment:
Why did you remove this?

Reply:
I moved it to right after the source is created (line 31). It changes the behavior by setting the filter before calling `Init`, but I checked the source and the order doesn't matter.